PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java

```diff
@@ -69,6 +72,36 @@ public String pluginVersion() {
         return currentConfiguration.pluginVersion();
     }
 
+    public void saveAzConfig() {
+        // from config -> state
+        final AzureConfiguration config = Azure.az().config();
+        final AzureConfigurations.AzureConfigurationData state = getState();
+        state.allowTelemetry(config.getTelemetryEnabled());
+        state.installationId(config.getMachineId());
+        state.environment(config.getCloud());
+        state.functionCoreToolsPath(config.getFunctionCoreToolsPath());
+        state.passwordSaveType(config.getDatabasePasswordSaveType());
+        state.pluginVersion(config.getVersion());
+    }
+
+    public boolean loadToAzConfig() {
+        // from state -> config
+        final AzureConfiguration config = Azure.az().config();
+        final AzureConfigurations.AzureConfigurationData state = getState();
+        if (StringUtils.isNotBlank(state.installationId())) {
+            config.setTelemetryEnabled(state.allowTelemetry());
+            config.setMachineId(state.installationId());
+            config.setCloud(state.environment());
+
+            if (StringUtils.isNotBlank(state.functionCoreToolsPath()) && new File(state.functionCoreToolsPath()).exists()) {
+                config.setFunctionCoreToolsPath(state.functionCoreToolsPath());
+            }
+            config.setDatabasePasswordSaveType(state.passwordSaveType());
```

Comment:
function core tools path and password save type should out of `if (StringUtils.isNotBlank(state.installationId())) {`. right ?